legend-depot-store-notifications/src/main/java/org/finos/legend/depot/store/notifications/store/mongo/NotificationsQueueMongo.java

```diff
@@ -113,15 +116,23 @@ public List<MetadataNotification> pullAll()
     }
 
     @Override
-    public Optional<MetadataNotification> getFirstInQueue()
+    public Optional<MetadataNotification> getFirstInQueueByPriority()
     {
-        Document first = (Document)getCollection().findOneAndDelete(Filters.exists("_id"),new FindOneAndDeleteOptions().sort(Sorts.ascending("createdAt")));
+        Document first = (Document)getCollection().findOneAndDelete(Filters.and(Filters.exists("_id"), Filters.eq("eventPriority", EventPriority.HIGH.name())), new FindOneAndDeleteOptions().sort(Sorts.ascending("createdAt")));
```

Comment:
does the deletion require sorting by createdAt, would it make sense to have First Come First Serve with priority in accordance?

Reply:
your implementation is wrong it would not work for items with low priority
yes u need to get the item out of the queue so the sort it what gives you who is next... this query is saying give me the first one sorted by priority then creation date. so we have two with same priority it need to pick up the oldest
you need to change the sort options adn the